checker/tests/resourceleak/IndexMode.java

```diff
@@ -0,0 +1,17 @@
+// A test for a new false positive issued in release 3.36.0 but not 3.35.0.
+// Reported as part of https://github.com/typetools/checker-framework/issues/6077.
+
+import java.util.Map;
+
+public class IndexMode {
+  public static Object getMode(Map<String, String> indexOptions) {
+    // Try-catch is needed, otherwise no FP.
```

Comment:
Do we understand why we need a try-catch and non-void return type here to reproduce the issue?  It seems to me the error should have just appeared with the `get()` call, since I think `literalOption` would have had the `@MustCall` type `TOP` before.

Reply:
Any thoughts on this comment @kelloggm?

Reply:
Sorry! I totally forgot about responding to this because I was focused on writing the new test. 

The answer is that we do not understand why. I have a theory, but I'm not that confident: I think maybe the consistency checker is only getting to the point of needing to do this check if there is exceptional control flow. The consistency checker does also reason about whether methods have return types (because of the need to possibly check obligations resolved), so I assume that there is some set of dependencies/coupling there. But, I'm definitely a bit worried by the fact that changing either of those things caused this test to no longer fail: it suggests that something funky is going on and probably ought to be investigated further.

Reply:
Can we add a similar test where there is an explicit @MustCall obligation on the `String` values for the `Map` (like `Map<String, @MustCall("hashCode") String>`), and then see if we get an error if the method body is just `String literalOption = indexOptions.get("is_literal");`?

Reply:
As we discussed earlier, I've settled this on leaving the false negative. I've added an explanation to the test case, and also added a test with a type that is expected to have a must-call obligation (`InputStream`), so this should be ready for re-review.